x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/components/url_drilldown_collect_config/url_drilldown_collect_config.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useRef, useState } from 'react';
+import {
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFormRow,
+  EuiIcon,
+  EuiLink,
+  EuiPopover,
+  EuiCheckbox,
+  EuiText,
+  EuiTextArea,
+} from '@elastic/eui';
+import { UrlDrilldownConfig, UrlDrilldownScope } from '../../types';
+import { compile } from '../../url_template';
+import { validateUrlTemplate } from '../../url_validation';
+import { buildScopeSuggestions } from '../../url_drilldown_scope';
+import './index.scss';
+import {
+  txtAddVariableButtonTitle,
+  txtUrlTemplateHelpLinkText,
+  txtUrlTemplateLabel,
+  txtUrlTemplateOpenInNewTab,
+  txtUrlTemplatePlaceholder,
+  txtUrlTemplatePreviewLabel,
+  txtUrlTemplatePreviewLinkText,
+} from './i18n';
+
+export interface UrlDrilldownCollectConfig {
+  config: UrlDrilldownConfig;
+  onConfig: (newConfig: UrlDrilldownConfig) => void;
+  scope: UrlDrilldownScope;
+  syntaxHelpDocsLink?: string;
+}
+
+export const UrlDrilldownCollectConfig: React.FC<UrlDrilldownCollectConfig> = ({
+  config,
+  onConfig,
+  scope,
+  syntaxHelpDocsLink,
+}: UrlDrilldownCollectConfig) => {
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const urlTemplate = config.url.template ?? '';
+  const compiledUrl = React.useMemo(() => {
+    try {
+      return compile(urlTemplate, scope);
+    } catch {
+      return urlTemplate;
+    }
+  }, [urlTemplate, scope]);
+  const scopeVariables = React.useMemo(() => buildScopeSuggestions(scope), [scope]);
+
+  function updateUrlTemplate(newUrlTemplate: string) {
+    if (config.url.template !== newUrlTemplate) {
+      onConfig({
+        ...config,
+        url: {
+          ...config.url,
+          template: newUrlTemplate,
+        },
+      });
+    }
+  }
+  const { error, isValid } = React.useMemo(
+    () => validateUrlTemplate({ template: urlTemplate }, scope),
+    [urlTemplate, scope]
+  );
+  const isEmpty = !urlTemplate;
+  const isInvalid = !isValid && !isEmpty;
+  return (
+    <>
```

Comment:
This is just a guess. Maybe there is something like `EuiForm` component to wrap around forms.

```suggestion
    <EuiForm>
```

Reply:
afaik, html forms shouldn't be nested (we have a form on a root of drilldown manager)